src/zenml/models/v2/base/base.py

```diff
@@ -94,28 +80,40 @@ class BaseResponseMetadata(BaseZenModel):
     """
 
 
+class BaseResponseResources(BaseZenModel):
+    """Base resources model.
+
+    Used as a base class for all resource models associated with responses.
+    """
+
+    class Config:
+        """Allows additional resources to be added."""
+
+        extra = Extra.allow
+
+
 AnyBody = TypeVar("AnyBody", bound=BaseResponseBody)
 AnyMetadata = TypeVar("AnyMetadata", bound=BaseResponseMetadata)
+AnyResources = TypeVar("AnyResources", bound=BaseResponseResources)
 
 
-class BaseResponse(GenericModel, Generic[AnyBody, AnyMetadata], BaseZenModel):
-    """Base domain model."""
-
-    id: UUID = Field(title="The unique resource id.")
-    permission_denied: bool = False
```

Comment:
why not keep `permission_denied` in the base class ? we'll probably need to run permission checks for in-memory models too.

Reply:
Had a discussion about this with Michael - for the time being no, no permission checks there. The rbac implementation also runs with the assumption that resources have ids. If we want to enable permission checks for in-memory items in the future, we can move it then